smarts/core/sensors.py

```diff
@@ -862,7 +862,7 @@ def append_waypoint_if_new(self, new_wp):
             or wp_edge in self._mission_planner.route.edges
         )
 
-        if most_recent_wp != new_wp and should_count_wp:
+        if np.linalg.norm(new_wp.pos - most_recent_wp.pos) > 0.5 and should_count_wp:
```

Comment:
.5 assumes distance is 1 unit? but it's actually parameterizable right?

Reply:
With the changes to the equally spaced waypoints, the waypoint spacing parameter does not have as much of an effect since the waypoints move around with the vehicle.

0.5 is a reasonable approximation to the old behaviour and should behave consistently across different spacings as well

Reply:
Is this not saying: "if we're passed the mid point..."?

Reply:
This change is saying "is the current waypoint we are holding more than 0.5 meters from the last waypoint that I recorded, if so, record this waypoint".